### PR TITLE
Fix <emails> in notifications 2

### PR DESCRIPTION
**Pull request by AdrienClairembault**
Following #8330, \<emails\> were correctly escaped in the mail queue but were still not displayed when receiving the actual email.
I figured there must be yet another HTML decoding step taking place before sending the mail, found it in notificationeventmailing and removed it.

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !21113

### CODE REVIEW

**Comment by trasher** [comment]
Won't this change cause other issues? I'm never confident with sanitize issues.

**Comment by cedric-anne** [comment]
Issue is the same as the one that affect presence of `<` char in rich text, see https://github.com/glpi-project/glpi/pull/6111 .
Problem here is chaining of `Toolbox::unclean_cross_side_scripting_deep()` (called in `NotificationTemplate::process()`), and `Html::entity_decode_deep()` that you removed.

I disagree with this PR as we may have legitimate HTML entities that should be decoded (maybe on when locales or data contains special chars).

**Comment by AdrienClairembault** [comment]
Then we can keep the decoding but double encode the <emails> again (like in the previous PR).
Feels hacky but it works.

File: inc/notificationtargetticket.class.php [hunk 1]
```diff
@@ -170,9 +170,7 @@ function getDataForObject(CommonDBTM $item, array $options, $simple = false) {
       // Without double encoding, these emails are interpreted as html and not
       // rendered in the final mail
       $regex = "/(&lt;[^;]+?@[^;]+?&gt;)/";
```
Comment (cedric-anne) [inline inc/notificationtargetticket.class.php:172]:
```suggestion
```

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**